ifsbench/config_mixin.py

```diff
@@ -0,0 +1,36 @@
+from typing import Dict, List, Union
+
+from pydantic import BaseModel, model_validator
+from typing_extensions import Self
+
+__all__ = ['ConfigMixin', 'CLASSNAME', 'RESERVED_NAMES']
+
+CLASSNAME = 'classname'
+RESERVED_NAMES = [
+    CLASSNAME,
+]
+
+
+class ConfigMixin(BaseModel):
+
+    @classmethod
+    def from_config(
+        cls, config: Dict[str, Union[str, float, int, bool, List, None]]
+    ) -> 'ConfigMixin':
+        return cls(**config)
+
+    def dump_config(
+        self, with_class: bool = False
+    ) -> Dict[str, Union[str, float, int, bool, List]]:
```

Comment:
Isn't `Union[str, float, int, bool, List]` not practically `Any`?

Reply:
In practice probably yes. But I wanted to make it explicit that these have to be types that can be supported when then dumping to and reading from yaml/toml/json.
After Johannes' requested change to make the NamelistHandler's input_/output_path Path objects, I actually added special handling to convert Path to str. I had originally intended to leave that to the yaml dumper, but then we'd have to reimplement for that for every supported format. 
So the output here should be whatever yaml can cleanly put into the config output and can then be read back and used again.